cmake/sysbuild/modules/ncs_sysbuild_extensions.cmake

```diff
@@ -5,6 +5,78 @@
 
 include_guard(GLOBAL)
 
+# Usage:
+#   UpdateableImage_Add(APPLICATION <name>
+#                       GROUP <name>
+#   )
+#
+# This function includes an image inside a list that will be used to trigger
+# bootloader-dependent build and packaging logic.
+#
+# APPLICATION <name>: Name of the application.
+# GROUP <name>:       Application group (i.e. VARIANT)
+function(UpdateableImage_Add)
+  cmake_parse_arguments(VIMAGE "" "APPLICATION;GROUP" "" ${ARGN})
+
+  if(VIMAGE_GROUP)
+    set(group ${VIMAGE_GROUP})
+  else()
+    set(group "DEFAULT")
+  endif()
+
+  # Update list of avilable groups
+  get_property(
+    groups GLOBAL PROPERTY sysbuild_updateable_groups
+  )
+  list(APPEND groups ${group})
+  list(REMOVE_DUPLICATES groups)
+  set_property(
+    GLOBAL PROPERTY sysbuild_updateable_groups ${groups}
+  )
+
+  # Append the image name
+  set_property(
+    GLOBAL
+    APPEND PROPERTY sysbuild_updateable_${group} ${VIMAGE_APPLICATION}
+  )
+endfunction()
+
+# Usage:
+#   UpdateableImage_Get(<outvar> <group>)
+#
+# This function returns a list of images, assigned to the specified group.
+#
+# <outvar>: Name of variable to set.
+# <group>:  Image update group.
+#           Use "DEFAULT" to get the default group (including DEFAULT_IMAGE).
+#           Use "ALL" to get images from all groups.
+function(UpdateableImage_Get outvar group)
+  set(all_images )
+  if(${group} STREQUAL "ALL")
+    get_property(
+      group GLOBAL PROPERTY sysbuild_updateable_groups
+    )
+    list(APPEND group "DEFAULT")
+    list(REMOVE_DUPLICATES group)
+  endif()
+
+  foreach(image_group ${group})
+    get_property(
+      images GLOBAL PROPERTY sysbuild_updateable_${image_group}
+    )
+    if(images)
+      list(APPEND all_images "${images}")
+    endif()
+
+    # Include the DEFAULT_IMAGE only if a bootloader is enabled.
+    if(${image_group} STREQUAL "DEFAULT" AND (NOT SB_CONFIG_BOOTLOADER_NONE))
+      list(APPEND all_images ${DEFAULT_IMAGE})
```

Comment:
not sure why this only includes the default image if a bootloader is enabled? It also does not match up with the description above